pkg/kubectl/cmd/logs.go

```diff
@@ -199,7 +225,40 @@ func (o LogsOptions) Validate() error {
 
 // RunLogs retrieves a pod log
 func (o LogsOptions) RunLogs() (int64, error) {
-	req, err := o.LogsForObject(o.Object, o.Options)
+	switch t := o.Object.(type) {
```

Comment:
I think I only now realized what you're speaking about. Still the biggest problem here is that depending on whether you use the selector or now you're getting different results. Since only `PodList` will trigger getting logs from all the containers, whereas single `Pod` will only return the first container (if none is specified). We need to make this consistent, I'm trying to come up with some reasonable solution.